app/src/main/java/com/demo/mdb/pokedex/MainActivity.java

```diff
@@ -0,0 +1,154 @@
+
+        final ListViewAdapter listPokemonAdapter = new ListViewAdapter(getApplicationContext(), pokemonArrayList);
+        final GridViewAdapter gridPokemonAdapter = new GridViewAdapter(getApplicationContext(), pokemonArrayList);
+        recyclerView.setAdapter(listPokemonAdapter);
+
+        lowSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
+                low = b;
+                listPokemonAdapter.pokemonArrayList = updatedArrayList();
+                listPokemonAdapter.notifyDataSetChanged();
+                gridPokemonAdapter.pokemonArrayList = updatedArrayList();
+                gridPokemonAdapter.notifyDataSetChanged();
+            }
+        });
+
+        medSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
+                med = b;
+                listPokemonAdapter.pokemonArrayList = updatedArrayList();
+                listPokemonAdapter.notifyDataSetChanged();
+                gridPokemonAdapter.pokemonArrayList = updatedArrayList();
+                gridPokemonAdapter.notifyDataSetChanged();
+            }
+        });
+
+        highSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
+                high = b;
+                listPokemonAdapter.pokemonArrayList = updatedArrayList();
+                listPokemonAdapter.notifyDataSetChanged();
+                gridPokemonAdapter.pokemonArrayList = updatedArrayList();
+                gridPokemonAdapter.notifyDataSetChanged();
+            }
+        });
+
+        button.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                isGridView = !isGridView;
+                //ArrayList<Pokedex.Pokemon> list = listPokemonAdapter.pokemonArrayList;
+                if (isGridView) {
+                    recyclerView.setLayoutManager(new GridLayoutManager(getApplicationContext(), 2));
+                    //pokemonAdapter = new GridViewAdapter(getApplicationContext(), list);
+                    recyclerView.setAdapter(gridPokemonAdapter);
+                } else {
+                    recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+                    //PokemonAdapter = new ListViewAdapter(getApplicationContext(), list);
+                    recyclerView.setAdapter(listPokemonAdapter);
+                }
+            }
+        });
+    }
+
+    public ArrayList<Pokedex.Pokemon> updatedArrayList()
+    {
+        ArrayList<Pokedex.Pokemon> updatedPokemonArrayList = new ArrayList<Pokedex.Pokemon>(pokemonArrayList);
```

Comment:
It's a lot more efficient to store indexes that point to singular copies of pokemon instead of cloning them every time you put them in a different place. It's also pretty unethical to just clone pokemon like that, cmon.